### PR TITLE
[RFC]vim-patch:8.2.0295: highlighting for :s wrong when using different separator

### PR DESCRIPTION
**Pull request by glacambre**

### CODE REVIEW

**Comment by janlazo** [comment]
@glacambre Can you verify if there are later patches that improve or fix this patch?

File: src/nvim/search.c [hunk 1]
```diff
@@ -1101,8 +1102,8 @@ int do_search(
 
     searchstr = pat;
     dircp = NULL;
-    /* use previous pattern */
-    if (pat == NULL || *pat == NUL || *pat == dirc) {
+    // use previous pattern
+    if (pat == NULL || *pat == NUL || *pat == search_delim) {
```
Comment (janlazo) [inline src/nvim/search.c:1106]: You missed some changes in this function.

File: src/nvim/testdir/test_search.vim [hunk 2]
```diff
@@ -723,6 +723,30 @@ func Test_incsearch_substitute_dump()
   call delete('Xis_subst_script')
 endfunc
 
+func Test_incsearch_highlighting()
```
Comment (janlazo) [inline src/nvim/testdir/test_search.vim:726]: Port this toa lua screen test but keep this for easier merging.

Comment (glacambre) [reply src/nvim/testdir/test_search.vim:726]: I recycled the test from https://github.com/neovim/neovim/pull/14215 - funnily enough while the vim test didn't catch the missing updates in screen.c my test did.

File: src/nvim/search.c (hunk 1 above)
Comment (glacambre) [reply src/nvim/search.c:1106]: Indeed, thank you for catching that!

**Review by glacambre — COMMENTED** [review]
Thank you for reviewing my PR :).

AFAICS, there aren't any other patches related to this bug.

**Comment by glacambre** [comment]
I think I might have missed yet another hunk. Please wait before merging this.

**Comment by glacambre** [comment]
Yes, I had botched the application of one hunk. Things should be hopefully ok now.

**Review by janlazo — APPROVED** [review]

**Pull request merged**